[system-probe] exit if debugfs is not mounted

shang-wang opened the pull request:
### What does this PR do?

Quit running system-probe with message to remind user to mount debugfs to `/sys/kernel/debug` in order to get it running. cc: @DataDog/burrito 

### Motivation

Previously the default message wasn't intuitive enough to help the user to fix this and would cause the system-probe to crash

### Additional Notes

I used the existence of `/sys/kernel/debug/tracing/kprobe_events` to this test, since it's the first file needed when enabling kprobe.

shang-wang committed:
[system-probe] check debugfs mount and exit if not mounted

shang-wang committed:
Merge branch 'master' into shang/mountfs-check

shang-wang committed:
[proc] move platform check and debugfs mount check before systemprobe
definition

shang-wang committed:
[proc] use existence of /sys/kernel/debug/tracing/kprobe_events to test
if debugfs is mounted or not

sunhay commented on cmd/system-probe/main.go:
```diff
@@ -93,6 +93,19 @@ func main() {
+
+	mounted := util.IsDebugfsMounted()
+	if !mounted {
+		log.Info("debugfs is not mounted, run \"sudo mount -t debugfs none /sys/kernel/debug\" to mount debugfs")
```
I think it would be useful to add a little more context: "debugfs is not mounted and is needed for eBPF-based checks, ..."

sunhay replied:
Hmm - should can this be done where ever the OS unsupported check is done? I think the order we want is OS check and then debugfs check.

sunhay commented on pkg/process/config/config.go:
```diff
@@ -347,7 +347,6 @@ func loadEnvVariables() {
 		"DD_DISABLE_IPV6_TRACING":   "system_probe_config.disable_ipv6",
 		"DD_COLLECT_LOCAL_DNS":      "system_probe_config.collect_local_dns",
 		"DD_USE_LOCAL_SYSTEM_PROBE": "system_probe_config.use_local_system_probe",
-		"DD_ENABLE_PROFILING":       "system_probe_config.debug_profiling_enabled",
```
Ah - good catch.

shang-wang committed:
[proc] better context on info

shang-wang committed:
[proc] check debugfs mounted after kernel version check

shang-wang committed:
move debugfs check to CreateSystemProbe

shang-wang committed:
move the platform info back
Originally I thought it was for linux version check but I misunderstood
the meaning of util.GetPlatform()

sunhay commented on pkg/process/util/util.go:
```diff
@@ -112,6 +112,14 @@ func GetPlatform() (string, error) {
 
+// IsDebugfsMounted would test the existence of file /sys/kernel/debug/tracing/kprobe_events to determine if debugfs is mounted or not
+func IsDebugfsMounted() bool {
+	if _, err := os.Stat("/sys/kernel/debug/tracing/kprobe_events"); err == nil {
```
```
_, err := os.Stat("/sys/kernel/debug/tracing/kprobe_events")
return err == nil
```

shang-wang committed:
[proc] nit
